etna/transforms/timestamp/holiday.py

```diff
@@ -83,14 +141,32 @@ def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         -------
         :
             pd.DataFrame with added holidays
+
+        Raises
+        ------
+        ValueError:
+            if the frequency is greater than daily and this is a ``binary`` or ``categorical`` mode
+        ValueError:
+            if the frequency is less than daily and this is ``days_count`` mode
         """
-        if (df.index[1] - df.index[0]) > datetime.timedelta(days=1):
-            raise ValueError("Frequency of data should be no more than daily.")
+        if bigger_than_day(self.freq) and self._mode is not HolidayTransformMode.days_count:
+            raise ValueError("For binary and category modes frequency of data should be no more than daily.")
 
         cols = df.columns.get_level_values("segment").unique()
         out_column = self._get_column_name()
 
-        if self._mode is HolidayTransformMode.category:
+        if self._mode is HolidayTransformMode.days_count:
+            numbers_part = "".join(itertools.takewhile(str.isdigit, self.freq))
+            letters_part = "".join(itertools.dropwhile(str.isdigit, self.freq))
+            numbers_part = "1" if numbers_part == "" else numbers_part
+            encoded_matrix = np.empty(0)
+            for dt in df.index:
+                start_date, end_date = define_period(letters_part[0], dt, numbers_part, self.freq)
+                date_range = pd.date_range(start=start_date, end=(end_date - timedelta(days=1)), freq="d")
+                count_holidays = sum(1 for d in date_range if d in self.holidays)
+                holidays_freq = count_holidays / date_range.size
+                encoded_matrix = np.append(encoded_matrix, holidays_freq)
+        elif self._mode is HolidayTransformMode.category:
             encoded_matrix = np.array(
                 [self.holidays[x] if x in self.holidays else self._no_holiday_name for x in df.index]
             )
```

Comment:
Let's also add `elif` on `HolidayTransformMode.binary` and in `else` put `assert_never(self._mode)`.